compositor_pipeline/src/pipeline/encoder.rs

```diff
@@ -257,17 +264,33 @@ where
         ctx: Arc<PipelineCtx>,
         options: Encoder::Options,
         source: Source,
-    ) -> Result<(Self, AudioEncoderConfig), EncoderInitError> {
+    ) -> Result<(Self, AudioEncoderStreamContext), EncoderInitError> {
+        let (packet_loss_sender, packet_loss_receiver) = watch::channel(0);
         let (encoder, config) = Encoder::new(&ctx, options)?;
+
         Ok((
             Self {
                 encoder,
                 source,
+                packet_loss_receiver,
                 eos_sent: false,
             },
-            config,
+            AudioEncoderStreamContext {
+                packet_loss_sender,
+                config,
+            },
         ))
     }
+
+    fn read_packet_loss(&mut self) -> Option<i32> {
```

Comment:
```suggestion
    fn updated_packet_loss(&mut self) -> Option<i32> {
```
old name implies that it always returns the value